recipe/meta.yaml

```diff
@@ -24,14 +24,13 @@ requirements:
   host:
     - python
     - numpy
-    - pybind11 >=2.8.0
+    - pybind11 >=2.12.0
```

Comment:
@jakirkham, out of curiosity, why the constraint?

https://github.com/pyamg/pyamg/blob/v5.2.1/pyproject.toml#L6

Reply:
Ah, I see it in your commit message now.

Reply:
good to know!

Reply:
Because `pybind11` did not have NumPy 2 support until `pybind11` version `2.12.0` ( https://github.com/pybind/pybind11/issues/5009 )

Agreed was just writing up this issue when you commented 😅 : https://github.com/pyamg/pyamg/issues/415